### ISSUE TITLE
Separate monitoring module

### ISSUE THREAD

**Issue by tvlkmirzawidihananta** [tvlk-data/rm-cookiecutter-serving#24]
Current monitoring implementation is in the cookiecutter template, and it will be copy-pasted in different serving repo. IMHO it will make the update difficult in the future. We should separate it into another library. WDYT?

**Comment by tvlkmirzawidihananta** [comment]
We already developed [ml-common](https://github.com/tvlk-data/ml-common) anyway. It has a nice feature like:
- tracking
- monitoring
- model loading in various platform
overall it want to make the inference service flow unified and easier

how about integrate it with RMI, and adding the stackdriver monitoring in that library?

**Comment by eduardchai** [comment]
Yes, i discussed the state of ml-common with mas @arinto before and checked the direction of the it. From what i got from mas @arinto, ml-common is not programmed to be library in the first place and there is initiative to make a proper ml libs. And he seems to favor more independent libs structure than the big super libs (like ml-common). Example:
```
# instead of:
from ml_common import Track, Monitor

# it will be 2 separate libs:
import ml_common_track
import ml_common_monitor
```

Because this is initiative from team NVS, can you share what has been discussed internally regarding this new common libs initiative?

**Comment by tvlkmirzawidihananta** [comment]
Hmmm, actually I know only the plan to package it into a proper library ([here is the discussion about it](https://github.com/tvlk-data/ml-common/issues/53))
Maybe mas @arinto can explain more the further plans here ^^a

**Comment by tvlkmirzawidihananta** [comment]
Btw, do you guys also willing to contribute to developing the ml-common? 😬
